test/unit/player.test.js

```diff
@@ -1044,3 +1044,21 @@ test('When VIDEOJS_NO_DYNAMIC_STYLE is set, apply sizing directly to the tech el
   equal(player.tech_.el().width, 600, 'the width is equal to 600');
   equal(player.tech_.el().height, 300, 'the height is equal 300');
 });
+
+test('When Android Chrome reports Infinity duration with currentTime 0, return NaN', function() {
+  const oldIsAndroid = browser.IS_ANDROID;
+  const oldIsChrome = browser.IS_CHROME;
+  const player = TestHelpers.makePlayer();
+
+  browser.IS_ANDROID = true;
+  browser.IS_CHROME = true;
+
+  player.tech_.el_ = {
+    duration: Infinity,
+    currentTime: 0
+  };
+  ok(isNaN(player.tech_.duration()), 'returned NaN with currentTime 0');
```

Comment:
We should use `Number.isNaN` instead. It's more correct and actually does what you expect it to do.